lib/srfi/128/comparators.scm

```diff
@@ -1,12 +1,26 @@
 
 (define-record-type Comparator
-  (make-comparator type-test equality ordering hash)
+  (%make-comparator% type-test equality ordering hash)
   comparator?
   (type-test comparator-type-test-predicate)
   (equality comparator-equality-predicate)
   (ordering comparator-ordering-predicate)
   (hash comparator-hash-function))
 
+(define (make-comparator type-test equality ordering hash)
+  (%make-comparator%
+    type-test
+    equality
+    ordering
+    (if (or (opcode? hash)
+            (not (procedure? hash))
+            (procedure-variadic? hash)
+            (not (= (procedure-arity hash) 1)))
+        hash
+        (case-lambda
```

Comment:
`case-lambda` is rather inefficient, and in this particular case `(lambda (x . o) (hash x))` is clearer anyway.

Reply:
Done.